duckduckgo/BrowserServicesKit
# Support customUserAgent

Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/1200890834746050/1200932417493197/f
iOS PR: https://github.com/duckduckgo/iOS/pull/1142
macOS PR: https://github.com/more-duckduckgo-org/macos-browser/pull/567
What kind of version bump will this require?: Patch

**Optional**:

Tech Design URL:
CC: @brindy 

**Description**:
This PR adds `customUserAgent` to the list of supported config features.

**Tagging instructions**
If this PR isn't ready to be merged for whatever reason it should be marked with the `DO NOT MERGE` label (particularly if it's a draft)
If it's pending Product Review/PFR, please add the `Pending Product Review` label.

If at any point it isn't actively being worked on/ready for review/otherwise moving forward (besides the above PR/PFR exception) strongly consider closing it (or not opening it in the first place). If you decide not to close it, make sure it's labelled to make it clear the PRs state and comment with more information.

**Steps to test this PR**:
1. Ensure you're using the embedded config by modifying PrivacyConfigManager 
2. On iOS: Visit apple.com
1. Log `navigator.userAgent` to the console and ensure it contains "DuckDuckGo"
4. Visit cvs.com
5. `navigator.userAgent` should **NOT** contain "DuckDuckGo"
6. macOS should be unaffected by this change



**OS Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
